8 reviews endpoints

ToastOfToastyness opened the pull request:
Resolves #8 

absolute OpenApi noob request review for reviews

Commit:
added query dictionary to schemas
added limit.yaml, messageBoard.yaml, review.yaml and reviewForm.yaml
added references to the new .yaml filed in _index.yaml

Commit:
added query dictionary to schemas
added limit.yaml, messageBoard.yaml, review.yaml and reviewForm.yaml
added references to the new .yaml filed in _index.yaml

Commit:
added query dictionary to schemas
added limit.yaml, messageBoard.yaml, review.yaml and reviewForm.yaml
added references to the new .yaml filed in _index.yaml

Commit:
Merge remote-tracking branch 'origin/8-reviews-endpoints' into 8-reviews-endpoints

Commit:
gestrige änderungen,die ohne vorwissen waren gelöscht

Commit:
gestrige änderungen,die ohne vorwissen waren gelöscht

Commit:
added parameters: messageBoardComment_id.yaml and review_id.yaml
added review endpoints: review.yaml, reviews.yaml and messageBoardComment.yaml
added schemas : messageBoardComment.yaml and reviewForm.yaml
edited schema review.yaml

Commit:
added parameters: messageBoardComment_id.yaml and review_id.yaml
added review endpoints: review.yaml, reviews.yaml and messageBoardComment.yaml
added schemas : messageBoardComment.yaml and reviewForm.yaml
edited schema review.yaml

Commit:
corrected typos and tabs

Commit:
corrected typos and tabs

bits0rcerer commented on src/openapi.yaml:
```diff
@@ -26,7 +26,17 @@ paths:
 
-# TODO: add paths here
+  #review endpoints
+  /reviews/reviews:
```
```suggestion
  /reviews:
```

bits0rcerer commented on src/openapi.yaml:
```diff
@@ -26,7 +26,17 @@ paths:
-# TODO: add paths here
+  #review endpoints
+  /reviews/reviews:
+    $ref: "./paths/reviews/reviews.yaml"
```
```suggestion
    $ref: "./paths/reviews.yaml"
```

bits0rcerer commented on src/openapi.yaml:
```diff
@@ -26,7 +26,17 @@ paths:
+    $ref: "./paths/reviews/reviews.yaml"
+
+  /reviews/{review_id}:
+    $ref: "./paths/reviews/review.yaml"
```
```suggestion
    $ref: "./paths/reviews/review_id/review.yaml"
```

bits0rcerer commented on src/openapi.yaml:
```diff
@@ -26,7 +26,17 @@ paths:
+  /reviews/{review_id}:
+    $ref: "./paths/reviews/review.yaml"
+
+  /reviews/{review_id}/{messageBoardComment_id}:
```
```suggestion
  /reviews/{review_id}/{comment_id}:
```

bits0rcerer commented on src/openapi.yaml:
```diff
@@ -26,7 +26,17 @@ paths:
+    $ref: "./paths/reviews/review.yaml"
+
+  /reviews/{review_id}/{messageBoardComment_id}:
+    $ref: "./paths/reviews/review_id/messageBoardComment.yaml"
```
```suggestion
    $ref: "./paths/reviews/review_id/comment_id/comment.yaml"
```

bits0rcerer commented on src/parameters/path/messageBoardComment_id.yaml:
```diff
@@ -0,0 +1,6 @@
+name: messageBoardComment_id
```
```suggestion
name: comment_id
```

bits0rcerer commented on src/paths/reviews/review.yaml:
```diff
@@ -0,0 +1,84 @@
+get:
+  tags:
+    - reviews
+  description: "Returns one review from an entry"
```
Where do one specify the entry?

bits0rcerer commented on src/paths/reviews/review.yaml:
```diff
@@ -0,0 +1,84 @@
+  requestBody:
+    description: "New PDF review"
+    required: true
+    content:
```
```suggestion
    content:
      application/pdf:
        schema:
          type: object
```

Content type missing.

bits0rcerer reviewed (changes requested):
Plesase stick to **_lower_snake_case_** naming.
Please change `message_board_comment`  to `comment`

Note: Do not just blindly apply suggestions as they do not fully apply, just roughly indicate the idea.

bits0rcerer commented:
Please keep commits english and in present term.

~~added ...~~ -> **adds ...**

ToastOfToastyness committed:
Update src/openapi.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

ToastOfToastyness committed:
Update src/openapi.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

ToastOfToastyness committed:
Update src/openapi.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

ToastOfToastyness committed:
Update src/openapi.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

ToastOfToastyness committed:
Update src/openapi.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

ToastOfToastyness committed:
Update src/parameters/path/messageBoardComment_id.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'origin/8-reviews-endpoints' into 8-reviews-endpoints

# Conflicts:
#	src/openapi.yaml
#	src/parameters/path/messageBoardComment_id.yaml

ToastOfToastyness committed:
Update src/paths/reviews/review.yaml

Co-authored-by: bits0rcerer <25325997+bits0rcerer@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'origin/8-reviews-endpoints' into 8-reviews-endpoints

Commit:
renames all instances of messageBoardComment and similar to just comment
moves comment.yaml into commment_id folder
moves review.yaml into review_id folder
moved reviews.yaml into paths folder
adjusts references of the moved files
renames reviewForm.yaml to review_form to fit lower_case_snake naming

bits0rcerer commented on src/paths/reviews.yaml:
```diff
@@ -0,0 +1,60 @@
+  tags:
+    - reviews
+  description: "Creates a review for an entry"
+  summary: "/createReview"
```
```suggestion
  summary: "/reviews"
```

bits0rcerer commented on src/paths/reviews.yaml:
```diff
@@ -0,0 +1,60 @@
+
+  requestBody:
+    description: "New review"
+    required: true
```
```suggestion
```
Kind of redundant

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+post:
+  tags:
+    - reviews
+    - messageBoard
```
```suggestion
```
i think it is enough to sort comments under reviews

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+    - reviews
+    - messageBoard
+  description: "Deletes a comment"
+  summary: "/<review_id>/<comment_id>"
```
```suggestion
  summary: "/reviews/<review_id>/<comment_id>""
```

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+    - reviews
+    - messageBoard
+  description: "Creates a new comment on the message board of a review"
+  summary: "reviews/<review_id>/<comment_id>"
```
```suggestion
  summary: "/reviews/<review_id>/<comment_id>"
```

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+
+  requestBody:
+    description: "New message board comment"
+    required: true
```
```suggestion
```

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+delete:
+  tags:
+    - reviews
+    - messageBoard
```
```suggestion
```

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+    - reviews
+    - messageBoard
+  description: "Changes the contents of the message board comment"
+  summary: "/<review_id>/<comment_id>"
```
```suggestion
  summary: "/reviews/<review_id>/<comment_id>"
```

bits0rcerer commented on src/paths/reviews/review_id/comment_id/comment.yaml:
```diff
@@ -0,0 +1,85 @@
+
+  requestBody:
+    description: "Changed contents of the comment"
+    required: true
```
```suggestion
```

bits0rcerer commented on src/paths/reviews/review_id/review.yaml:
```diff
@@ -0,0 +1,88 @@
+
+  requestBody:
+    description: "New PDF review"
+    required: true
```
```suggestion
```

bits0rcerer commented on src/schemas/review_form.yaml:
```diff
@@ -0,0 +1,20 @@
+type: object
```
Score and Confidence?
[See mocks](https://peerreq-mockup.bits0rcerer.org/categories/1/1)

bits0rcerer commented on src/schemas/review.yaml:
```diff
@@ -31,4 +31,14 @@ properties:
-    format: "ddd, dd. mmm yyyy"
\ No newline at end of file
+    format: "ddd, dd. mmm yyyy"
+  pdfReview:
```
lower_snake_case

Commit:
removes messageBpard tag
adjusts summaries of reviews.yaml and comment.yaml
removes requirement of requestBody in review.yaml and comment.yaml
adds references to review_form.yaml and comment.yaml in _index.yaml
comment.yaml: examples are actually strings now
adds score and confidence to review_form.yaml
adjusted review.yaml and review_form.yaml to fit lower_case_snake

bits0rcerer reviewed (changes requested):
I gave this some thought and came to the conclusion that it might be beneficial to have the following endpoints:

- `/reviews`:
  -  `GET`: Returns all reviews where the requesting user is either the researcher or the reviewer.
- `/reviews/{review_id}` ➜ let us remove this endpoint here. Any operation here on reviews might be better under the reviews context. 
- `/reviews/{review_id}/{comment_id}` ➜ also remove

I came to this conclusion that we only need a way to get reviews associated to the current user without any context regarding an entry. For every other Operation the requester has to know the entry. Therefore every operation on a specific review is suited best under the corresponding entry.

What do you think?

bits0rcerer commented on src/paths/reviews.yaml:
```diff
@@ -0,0 +1,59 @@
+  parameters:
+    - $ref: "../parameters/query/limit.yaml"
+    - $ref: "../parameters/query/after.yaml"
+    - $ref: "../parameters/path/entry_id.yaml"
```
A small thing I like to make clear. You cannot have a path parameter (`entry_id` in this case) that is not represented in the path (`/reviews` here).

For a valid path parameter it has to be in the path, otherwise there is no way for a requester to declare its value.

ToastOfToastyness replied:
Sorry, i didn't know!
My bad

bits0rcerer commented:
Let's conclude this PR with just the endpoint `/reviews`.
For any other operation on `reviews` let's discuss this in a different Issue.

See #18

bits0rcerer committed:
remove obsolete endpoints

bits0rcerer committed:
remove fixed scale from review_forms
